### ISSUE TITLE
How to Run the code in vscode

### ISSUE THREAD

**Issue by AchalaSB** [paritytech/wasmi#127]
How to run wasmi example (tictoctoe.rs) code??
I have done with cargo test everything is fine. Then I used cargo run to execute  .rs code.  But it not showing me any result.

What is the command to run code?

**Comment by pepyakin** [comment]
I'm not exactly sure how to run rust code in vscode, but here is how to run an example from the command-line:

```
cd path/to/wasmi
cargo run --example=<example name>
```

to run tictactoe example you will need two wasm files that uses certain interface (the file should export function named `mk_turn` and should use functions `get` `set` to manipulate the field). Once you have these files you can run the example like that:

```
cd path/to/wasmi
cargo run --example=tictactoe path/to/file1.wasm path/to/file2.wasm
```

Hope this helps!

**Comment by AchalaSB** [comment]
How the that wasm file is created? where should I found those?

**Comment by justinclift** [comment]
@AchalaSB You seem to be trying very hard to get Wasm going in either Rust or Go.

If you need a hand to get started - and having people to interact with directly would be useful - then the "WebAssembly" channel on the Gophers Slack is a good bet:

&nbsp; &nbsp; https://gophers.slack.com/messages/C4GU6CHSQ

Slack is technically "invite only", but don't worry.  There is a web page for sending out invites (for Gophers Slack) here:

&nbsp; &nbsp; https://invite.slack.golangbridge.org

That's the one I used, and it seems to work fine. :smile:

**Comment by AchalaSB** [comment]
@pepyakin  As you said I need two wasm file. Should I manually create those files? I did 
``` cargo build --target=wasm32-unknown-unknown```
and to connect with server run the command ```cargo web start --target=wasm32-unknown-unknown```
but code is not executing it shows blank page. 
How can I run tictactoe.rs in server?

**Comment by pepyakin** [comment]
The creation of these files are left as an exercise to the reader : ) I used to have these, but now I don't, sorry!

You see, wasmi is a standalone interpreter, it supposed to be used in embedding scenarios. E.g. image you're writing some image processing application, like Photoshop, and you need to implement plugin functionality, and you've decided to use wasm for it. You can integrate wasmi to support running wasm files in your app.

In other words, If you want to run wasm inside a browser you should definitely check out a great book about using rust with wasm https://rustwasm.github.io/book/  !

**Comment by zaynetro** [comment]
In case someone else would like to try out tictactoe example this is what you can do:

1. Create a new cargo project `cargo init --lib tictactoe-wasm`
2. Modify `Cargo.toml`

```toml
[package]
name = "tictactoe-wasm"
version = "0.1.0"
authors = ["John Smith <email@company.com>"]
edition = "2018"

[lib]
crate-type = ["cdylib"]
```

3. Write a naive implementation in `src/lib.rs`

```rust
extern {
    // Get which player that cell belongs to
    fn get(cell: i32) -> i32;
    // Mark cell as checked
    fn set(cell: i32);
}

#[no_mangle]
pub unsafe extern "C" fn mk_turn() {
    for cell in 0..9 {
        let value = get(cell);
        // If value is not set then check it
        if value == 0 {
            set(cell);
            return;
        }
    }
}
```

4. Compile to wasm

```
rustup target add wasm32-unknown-unknown
cargo build --target wasm32-unknown-unknown
# wasm files are now in `./target/wasm32-unknown-unknown/debug`
```

5. Go to wasmi repository and run tictactoe example

```
cd wasmi
cargo run --example tictactoe --release -- ../tictactoe-wasm/target/wasm32-unknown-unknown/debug/tictactoe_wasm.wasm ../tictactoe-wasm/target/wasm32-unknown-unknown/debug/tictactoe_wasm.wasm
```
